File: AISuite_QuickStart/app/src/main/java/com/zebra/aisuite_quickstart/java/detectors/warehouselocalizer/WareHouseLocalizerGraphic.java
package com.zebra.aisuite_quickstart.java.detectors.warehouselocalizer;

import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.graphics.Rect;

import com.zebra.aisuite_quickstart.GraphicOverlay;

import java.util.ArrayList;
import java.util.List;

public class WareHouseLocalizerGraphic extends GraphicOverlay.Graphic {
    private final Paint boxPaint;
    private final List<Rect> boundingBoxes = new ArrayList<>();
    public WareHouseLocalizerGraphic(GraphicOverlay overlay, List<Rect> boxes) {
        super(overlay);
        overlay.clear();

        // Initialize the paint for drawing bounding boxes
        boxPaint = new Paint();
        boxPaint.setColor(Color.GREEN);
        boxPaint.setStyle(Paint.Style.STROKE);
        boxPaint.setStrokeWidth(6f);

        boundingBoxes.clear();

        // Populate bounding boxes if provided
        if (boxes != null) {
            boundingBoxes.addAll(boxes);
        }



        // Trigger a redraw of the overlay
        postInvalidate();
    }
    @Override
    public void draw(Canvas canvas) {
        // Draw bounding boxes
        for (Rect rect : boundingBoxes) {
            canvas.drawRect(rect, boxPaint);
        }

    }
}

File: AISuite_QuickStart/app/src/main/java/com/zebra/aisuite_quickstart/kotlin/detectors/warehouselocalizer/WareHouseLocalizerGraphic.kt
package com.zebra.aisuite_quickstart.kotlin.detectors.warehouselocalizer

import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint
import android.graphics.Rect
import com.zebra.aisuite_quickstart.GraphicOverlay

class WareHouseLocalizerGraphic(overlay: GraphicOverlay,
                       boxes: List<Rect>?,) : GraphicOverlay.Graphic(overlay) {

    private val boxPaint: Paint = Paint().apply {
        color = Color.GREEN
        style = Paint.Style.STROKE
        strokeWidth = 6f
    }
    private val boundingBoxes: MutableList<Rect> = mutableListOf()
    init {
        overlay.clear()

        boxes?.let { boundingBoxes.addAll(it) }

        // Redraw the overlay, as this graphic has been added.
        postInvalidate()
    }

    /**
     * Draws the bounding boxes and decoded text on the given Canvas.
     *
     * @param canvas The canvas on which to draw the graphic.
     */
    override fun draw(canvas: Canvas) {
        for (rect in boundingBoxes) {
            canvas.drawRect(rect, boxPaint)
        }
    }
}